fastly/resource_fastly_ngwaf_datadog_alert_test.go

```diff
@@ -0,0 +1,181 @@
+		}
+		ws, ok := s.RootModule().Resources[workspaceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", workspaceName)
+		}
+		rID := rs.Primary.ID
+		wID := ws.Primary.ID
+
+		conn := testAccProvider.Meta().(*APIClient).conn
+		latest, err := ddalerts.Get(context.TODO(), conn, &ddalerts.GetInput{
+			AlertID:     &rID,
+			WorkspaceID: &wID,
+		})
+		if err != nil {
+			return fmt.Errorf("Unable to retrieve NGWAF Datadog Alert %s: %v", rID, err)
+		}
+
+		if latest == nil {
+			return fmt.Errorf("NGWAF Datadog Alert %s not found in API", wID)
+		}
+
+		*datadogAlertID = rID
+		*workspaceID = wID
+		log.Printf("[DEBUG] EXISTS IMPORT: NGWAF datadog alert ID: %s", *datadogAlertID)
+		log.Printf("[DEBUG] EXISTS IMPORT: NGWAF workspace ID: %s", *workspaceID)
+
+		return nil
+	}
+}
+
+func testAccCheckNGWAFDatadogAlertDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*APIClient).conn
+	var wsID string
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "fastly_ngwaf_workspace" {
+			continue
+		}
+
+		wsID = rs.Primary.ID
+	}
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "fastly_ngwaf_datadog_alert" {
+			continue
+		}
+
+		_, err := ddalerts.Get(context.TODO(), conn, &ddalerts.GetInput{
+			AlertID:     &rs.Primary.ID,
+			WorkspaceID: &wsID,
+		})
+		if err == nil {
+			return fmt.Errorf("NGWAF Datadog Alert %s still exists", rs.Primary.ID)
+		}
+	}
+	return nil
+}
+
+func testAccNGWAFDatadogAlertConfig(workspaceName string) string {
+	return fmt.Sprintf(`
+resource "fastly_ngwaf_workspace" "test_datadog_alert_workspace" {
+  name                         = "%s"
+  description                  = "Test NGWAF Workspace"
+  mode                         = "block"
+  ip_anonymization            = "hashed"
+  client_ip_headers           = ["X-Forwarded-For", "X-Real-IP"]
+  default_blocking_response_code = 429
+
+  attack_signal_thresholds {
+    one_minute  = 100
+    ten_minutes = 500
+    one_hour    = 1000
+    immediate   = true
+  }
+}
+
+resource "fastly_ngwaf_datadog_alert" "test_datadog_alert" {
+  description                  = "%s"
+  integration_key              = "%s"
+  integration_site             = "%s"
+  workspace_id                 = fastly_ngwaf_workspace.test_datadog_alert_workspace.id
+}
+`, workspaceName, datadogAlertDescription, datadogAlertKey, datadogAlertSite)
+}
+
+func testAccNGWAFDatadogAlertConfigUpdate(workspaceName string) string {
+	return fmt.Sprintf(`
+resource "fastly_ngwaf_workspace" "test_datadog_alert_workspace" {
+  name                         = "%s"
+  description                  = "Test NGWAF Workspace"
+  mode                         = "block"
+  ip_anonymization            = "hashed"
+  client_ip_headers           = ["X-Forwarded-For", "X-Real-IP"]
+  default_blocking_response_code = 429
+
+  attack_signal_thresholds {
+    one_minute  = 100
+    ten_minutes = 500
+    one_hour    = 1000
+    immediate   = true
+  }
+}
```

Comment:
This should be identical to the 'Create' version, so that there are no changes required in the workspace itself.